scico/optimize/admm.py

```diff
@@ -154,7 +154,7 @@ class LinearSubproblemSolver(SubproblemSolver):
             :math:`\mb{x}` update step.
     """
 
-    def __init__(self, cg_kwargs: Optional[dict] = None, cg_function: str = "scico"):
+    def __init__(self, cg_kwargs: Optional[dict[str, Any]] = None, cg_function: str = "scico"):
```

Comment:
In line 157 dict[str, Any] should be of type Dict of the typing module if I'm not mistaken.

Reply:
Check out https://mypy.readthedocs.io/en/stable/builtin_types.html, looks like @FernandoDavis is correct, `dict[stuff]` is a syntax error in Python 3.8.

Reply:
But it's supported natively in Python 3.9, and from Python 3.7 onward when using `from __future__ import annotations`, so I think this usage is fine (and note the absence of errors in the CI).